lib/spack/spack/solver/error_messages.lp

```diff
@@ -165,11 +165,10 @@ error(0, "Attempted to build package {0} which is not buildable and does not hav
 
 error(0, "Attempted to build package {0} which is not buildable and does not have a satisfying external\n        'Spec({0} {1}={2})' is an external constraint for {0} which was not satisfied\n        'Spec({0} {1}={3})' required", Package, Variant, Value, OtherValue, startcauses, OtherValueCause, CID)
   :- external(node(ID, Package)),
-     not attr("external_conditions_hold", node(ID, Package), _),
      imposed_constraint(EID, "external_conditions_hold", Package, _),
      trigger_and_effect(Package, TID, EID),
      condition_requirement(TID, "variant_value", Package, Variant, Value),
-     not attr("variant_value", node(ID, Package), Variant, Value),
+     attr("variant_value", node(ID, Package), Variant, Value),
      attr("variant_value", node(ID, Package), Variant, OtherValue),
```

Comment:
@becker33 on `develop`, it looks like when this error happens, we don't derive both variant values. But we *do* derive both for the other error I modified above (two values for a single-valued variant).  On this branch, we derive both values in both situations.  I don't exactly understand why -- I could dig deeper but thought I'd ask if you have any insights here.  Also, do we need tests and changes for any other error conditions?  Some of the others seem to fit this pattern and there are more error cases than tests.

Reply:
this is no longer needed -- not sure I understand why it wasn't working before, but the original encoding works again.